From 0d37614542d9ade840af6e0839ee2807eebae871 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 30 Dec 2021 07:39:18 +0200
Subject: [PATCH] [SW-68384] gaudi2: security: set axprot for psoc i2c_s

By default the permission level of i2c_s mstr if is
privilege therefore any transaction coming via i2c_s
is treated as privilege transaction. hence setting the
permission level to user level.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I235f71f6351b00019e4b1303b9667cd2b9febcab
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 6ff12ae..6a210d5 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -45,6 +45,7 @@
 #include <dcore0_vdec0_mstr_if_core_lbw_regs.h>
 #include <pmmu_hbw_mstr_if_core_hbw_regs.h>
 #include <jt_mstr_if_core_hbw_regs.h>
+#include <i2c_s_mstr_if_core_hbw_regs.h>
 #include "common.h"
 
 static struct hl_sec_rrs_cfg hl_sec_lbw_rrs_tbl[] = {
@@ -1142,6 +1143,8 @@
 {
 	/* Set jtag to user level */
 	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	/* Set i2c_s to user level */
+	regmap_writel(HL_SEC_AXPROT_USER_VAL, mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 }
 
 void preboot_security_init(void)
